### DISCUSSION TITLE
Request for Review: Version 1.85p, Proposed Program Improvements

### DISCUSSION THREAD

**Discussion by K6OK** [w7sst/MorseRunner#330] [Ideas]
Some time ago I looked at v1.84 with an eye towards improving the user interface and to make the program friendlier to new users.  Towards that end I developed this "beta" version.  Along the way Mike W7SST made a suggestion or two and tracked down a bug I couldn't find (thanks Mike!).

I'm looking for thoughts and feedback on this "beta" program.  Do you think it offer a better user experience or do you prefer the v1.84 style?  Please note this is not a v1.85 Release Candidate, only a prototype.  You may encounter bugs.

The main improvements are: Adds an "Activity" pulldown so user can specify if they are practicing for a contest, training, or competing (HST); adds a Pause function; makes entry fields larger; provide single click buttons for Run/Pause/Stop; put labels on RIT and the rate chart; provide all 12 F keys and a few other technical enhancements.  In the .zip file below there's a document that describes these more fully.

If you run the prototype, your comments and suggestions are welcome here in this thread.  Thanks!  73 Jim K6OK

[Download MRCE 1.85p](
https://drive.google.com/file/d/1OzbU53-DmegJC05SXiaETOnoSXDbj6_Y/view?usp=sharing)

@w7sst @f6fvy @scotthibbs @tekenny @DF9TS @BH1SCW @WR7Q @jr8ppg @w3us

**Comment by f6fvy** [comment]
Overall, I like the new layout, but I don't really understand the difference between training and practice. When choosing either option with the same contest and mode (PileUp / Single call), I don't see any difference, unless I've missed something (probably).

**Comment by jr8ppg** [comment]
Hi,
I also didn't understand the difference between practice and training.
I think the other improvements are good.

**Comment by K6OK** [comment]
Thank you both for your comments -- I'm glad you like the new layout!  Sorry about Practice and Training, I should have explained that better.

Practice = prepare for contest => show all available contests

Training = for students in CW classes, only show those contests the instructor is using.  For example, in the revised prototype (link below) the Training mode only shows CQ WPX and K1USN Slow Speed Test as the available options.  This is only an example.  In the future, might be able to support other default settings like lower speed, Farnsworth, etc., that change when Training is selected.  

Also, a comment was made that the arrows on the RIT control should be changed from < >  to v ^ .  This change will match the keyboard and mousewheel.

Here is the revised program if you wish to see it:  [Download MRCE1.85p0731](https://drive.google.com/file/d/1iRBNTMyD9iqHLAejpIAB1pEa2bq7Ptdz/view?usp=sharing)

@w7sst @f6fvy @jr8ppg @scotthibbs @WR7Q

**Comment by jr8ppg** [comment]
Hi K6OK san,
I understand about the training mode.
So in the future, you'll be able to set contests to be displayed in the training mode.

I think the RIT arrows are good.

**Reply by K6OK** [reply]
Hi JR8PPG -- thank you very much for your review and comments. 73

**Comment by WR7Q** [comment]
Hi Jim,

I finally got a moment to look at this, as we are in the rush of
registrations for next semester.

When I try to download 1.85p, I get a version that is dated 07/07/24. Is
this the one you want me to check out?

73,
Bob WR7Q
CWA Admin
8018796566

**Reply by K6OK** [reply]
Hi Bob.  That's an older one. Please download this one, dated 07/31: [MRCE1.85p0731](https://drive.google.com/file/d/1iRBNTMyD9iqHLAejpIAB1pEa2bq7Ptdz/view?usp=sharing)

73 Jim K6OK

**Comment by WR7Q** [comment]
Got it.Tnx

**Comment by WR7Q** [comment]
Hi Jim,

Sorry, it won't let me download.

73,
Bob WR7Q
CWA Admin

**Reply by K6OK** [reply]
Hi Bob.  Sorry you're having difficulties with the download.  Here's another link to try: [MRCE1.85p0731](https://filebin.net/fsqo6gss6qa5fs90)

That link expires in 7 days. Before downloading, you might want to temporarily turn off virus scanners if you use any.  
73 Jim

**Comment by WR7Q** [comment]
Hi Jim,

It downloaded that time. Tnx. I will look at it soon.

73,
Bob WR7Q
CWA Admin

**Comment by EA3CX** [comment]
Hi Jim and all,

I´m testing it and like it very much, think you have achieved your goals,
for now the only thing I have found is that it does not allow me to include
a letter in the zone when in CQWW ( I usually send 5NN A4).

Very nice layout, thanks

73
Ariel
EA3CX

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Libre
de virus.www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Reply by w7sst** [reply]
Hi Ariel,
Thank you for your feedback. I created a [bug report](https://github.com/w7sst/MorseRunner/issues/331) for the issue you mentioned regarding not being able to send `5NN A4` (see Issue #331). This will be easy to fix.

In the future, if you see any issues, please submit a bug report [here](https://github.com/w7sst/MorseRunner/issues/new/choose).

Again, we appreciate your feedback!

73,
Mike W7SST

**Comment by f6fvy** [comment]
May I suggest starting the pile up as soon as you press F1, once the settings (dropboxes etc.) have been made, without having to press "Run" with the mouse ?

**Reply by K6OK** [reply]
Yes, that would be convenient.  Your idea is a good candidate for an enhancement later.  --  Jim

**Comment by WR7Q** [comment]
Hi Jim,

So far I really like the changes you have made.

Only used CQ WPX and CWOPS CWT. I will check out ARRL FD in the next day or
so.

Thanks again.

73,
Bob WR7Q
CWA Admin

**Comment by f6fvy** [comment]
Regarding the key assignments, F12 was previously sending "NR?", and now it wipes everything, which is rather confusing ;-) I understand that there are plans to make this fully customizable, but in the meantime I suggest not changing anything so as not to confuse users (including me ;-) ). And historically, F11 was assigned to "Wipe QSO", not F12.

**Reply by K6OK** [reply]
Sorry, I didn't realize there was  a conflict.  I use N1MM and it uses F12 = Wipe, so that was the natural choice for me.  I'll change it back to F11.

**Comment by scotthibbs** [comment]
I love it! 

I appreciate you changing the RIT arrows to match the keyboard. 

Eventually it will be nice to have a keyboard shortcut for the RIT reset button too. 

Very well done!

**Reply by K6OK** [reply]
Thank you Scott.  Good idea with the keyboard shortcut for RIT reset.  I'm also thinking there needs to shortcuts for Run/Pause/Stop as well.

**Comment by w3us** [comment]
FWIW, I (and a few thousand others) use N1MM as well where F12 has always been the wipe key.  Thanks for a really nifty program.

Rusty - W3US

**Comment by f6fvy** [comment]
Use any key you like to Wipe, but we want "NR?" back ;-)

**Comment by K6OK** [comment]
Based on your suggestions, I have updated the prototype with the following changes:

F12 button or key = Wipe        (same as N1MM)
F9 button or key = send "NR?"   (same as N1MM)
(SHIFT + Up Arrow)  or (SHIFT + Down Arrow)  resets the RIT to zero.

Thanks for your review and feedback.  The updated file Morse Runner 1.85p0807.zip  is available at
[1.85p0807.zip](https://drive.google.com/file/d/1F7mD3q8zuDR0H5aKa2ff_jpsx7JBDRdo/view?usp=sharing)  

or  at   https://filebin.net/86tlc3zguvmezf59   (link expires in 7 days)    --   73 Jim

@f6fvy @scotthibbs @WR7Q @jr8ppg @w7sst @w3us

**Comment by f6fvy** [comment]
Excellent. Thanks.
